### PR TITLE
Add an Actions dependency.

### PR DESCRIPTION
**Pull request by wRAR**

### CODE REVIEW

File: docs/usage/scrapy-poet.rst [hunk 1]
```diff
@@ -93,10 +100,47 @@ country code:
...
+
+    from scrapy_zyte_api import Actions, actions_list
+
+
+    @attrs.define
+    class MyPageObject(BasePage):
+        product: Product
+        actions: Annotated[
+            Actions,
+            actions_list(
```
Comment (Gallaecio) [inline docs/usage/scrapy-poet.rst:130]: Could we call it `action_list` (no `s`) or `actions`?

**Review by Gallaecio — APPROVED** [review]
Awesome!

File: docs/usage/scrapy-poet.rst [hunk 2]
```diff
@@ -93,10 +100,47 @@ country code:
...
+                        "onError": "return",
+                    },
+                    {"action": "waitForTimeout", "timeout": 5, "onError": "return"},
+                ]
+            ),
+        ]
+
+You can access the results of these actions in the
+:attr:`.Actions.results` attribute of the dependency in the
+resulting page object.
```
Comment (kmike) [inline docs/usage/scrapy-poet.rst:146]: Could you please add a code example? it'd be nice to show e.g. how to check if action was a success or not.

Comment (kmike) [reply docs/usage/scrapy-poet.rst:146]: Maybe even in the validate_input method

File: scrapy_zyte_api/providers.py [hunk 3]
```diff
@@ -111,6 +113,11 @@ async def __call__(  # noqa: C901
                     raise ValueError("Geolocation dependencies must be annotated.")
                 zyte_api_meta["geolocation"] = cls.__metadata__[0]  # type: ignore[attr-defined]
                 continue
+            if cls_stripped is Actions:
+                if not is_typing_annotated(cls):
+                    raise ValueError("Actions dependencies must be annotated.")
```
Comment (kmike) [inline scrapy_zyte_api/providers.py:118]:
```suggestion
                    raise ValueError("Actions dependencies must be annotated, e.g. Annotated[Actions, action_list([...list of actions...])]")
```

File: scrapy_zyte_api/providers.py [hunk 4]
```diff
@@ -234,6 +241,18 @@ async def __call__(  # noqa: C901
...
+            if cls_stripped is Actions and is_typing_annotated(cls):
+                actions_result: Optional[List[_ActionResult]]
+                if "actions" in api_response.raw_api_response:
+                    actions_result = [
+                        _ActionResult(**action_result)
+                        for action_result in api_response.raw_api_response["actions"]
+                    ]
+                else:
+                    actions_result = None
+                item = AnnotatedInstance(Actions(actions_result), cls.__metadata__)  # type: ignore[attr-defined]
```
Comment (kmike) [inline scrapy_zyte_api/providers.py:253]: The variable name ("item") looks wrong; this applies to Geolocation dependency as well.

**Pull request merged**